phare/pp/diagnostic/patch.py

```diff
@@ -0,0 +1,58 @@
+class PatchLevel:
+    REFINEMENT_RATIO = 2
+    dim_to_xyz = {0: "x", 1: "y", 2: "z"}
+    xyz_to_dim = {"x": 0, "y": 1, "z": 2}
+
+    def __init__(self, diag, idx, cells):
+        self.diag, self.idx = diag, idx
+        power = pow(2, idx)
+        self.cells = [power * x for x in diag.sim.cells]
```

Comment:
no this works, cells argument is incorrect, to be assessed

Reply:
actually I think it does not make sense for a patch level to have `cells` since this in only valid for a rectangular contiguous domain and patch levels maybe a non rectangular, possibly disjoint, collection of patches.

I think it makes sense for patch levels to have cell width, and that may incidentally be through their reference to their patch_level that patches get the cell width (so that we are sure they all get the same), but `cells` only make sense for patches, not patchlevels

Reply:
you seemed content with 
```
self.amr_index_space_cells = [power * x for x in diag.cells]
```

no?

Reply:
`_cells` is unnecessary IMO, what do you think

Reply:
we use it for periodic particle overlap tests to find the max icell
can try move it to the test

Reply:
replaced with 

```
def max_amr_index_for(patch_level, direction):
    assert isinstance(patch_level, PatchLevel)

    return (
        REFINEMENT_RATIO ** patch_level.lvlNbr
        * patch_level.diag.cells[xyz_to_dim[direction]]
    )
```

Reply:
hmm ok